middleware/http/server.go

```diff
@@ -147,3 +162,228 @@ func (r *rwInterceptor) getStatusCode() int {
 func (r *rwInterceptor) getResponseSize() string {
 	return strconv.FormatUint(atomic.LoadUint64(&r.size), 10)
 }
+
+func (r *rwInterceptor) wrap() http.ResponseWriter {
```

Comment:
I trust you @basvanbeek and I believe this piece of code does its trick but what do you think about adding some documentation about what it does, why it works that way and why it is needed for the sake of clarity?

Reply:
It atomically reads value of r.size which returns a uint64 and that is converted to string using base 10 notation. 

I think we don't need atomicity here so I will remove that but rather do it in a future PR.